script.js

```diff
@@ -1,12 +1,138 @@
 //You can edit ALL of the code here
 function setup() {
   const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  const rootElem = document.getElementById("root");
+
+  // Create section below for header
+  let headerEl = document.createElement("header");
+  rootElem.appendChild(headerEl);
+  let sectionEl = document.createElement("section");
+  rootElem.appendChild(sectionEl);
+  setupHeader(headerEl);
+  setupSearchInput(headerEl, allEpisodes, sectionEl);
+  setupSelectEpisodes(allEpisodes, sectionEl);
+  makePageForEpisodes(allEpisodes, sectionEl);
+
+  setupPageCredits(rootElem);
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+function setupHeader(headerEl) {
+  // Create page title below
+  let pageTitle = document.createElement("h1");
+  pageTitle.innerText = "TV Show Project";
+  headerEl.appendChild(pageTitle);
+}
+
+function setupSearchInput(headerEl, episodeList, sectionEl) {
+  // Create search bar below.
+  let searchBar = document.createElement("div");
+  searchBar.setAttribute("id", "search__bar");
+  let searchInput = document.createElement("input");
+  searchInput.setAttribute("id", "input__field");
+  searchInput.type = "text";
+  searchInput.placeholder.innerHTML = "search";
+  let searchLabel = document.createElement("label");
+  searchLabel.setAttribute("id", "label");
+  let epCount = episodeList.length;
+  searchLabel.innerHTML = `Displaying <span id="epCount">${epCount}</span>/73 episodes`;
+
+  searchBar.appendChild(searchInput);
+  searchBar.appendChild(searchLabel);
+  headerEl.appendChild(searchBar);
+
+  searchInput.addEventListener("input", function () {
+    makePageForEpisodes(
+      episodeList.filter((episode) => {
+        let input = searchInput.value;
+        return (
+          episode.name.toUpperCase().includes(input.toUpperCase()) ||
+          episode.summary.toUpperCase().includes(input.toUpperCase())
+        );
+      }),
+      sectionEl
+    );
+  });
+}
+
+function setupSelectEpisodes(allEpisodes, sectionEl) {
+  let selectEl = document.createElement("select");
+  selectEl.setAttribute("id", "select__episodes");
+  let searchBar = document.getElementById("search__bar");
+  searchBar.appendChild(selectEl);
+
+  let selectOption = document.createElement("option");
+  selectOption.text = "Select for all episodes.";
+  selectEl.add(selectOption);
+
+  selectEl.addEventListener("change", function () {
+    if (this.value === "Select for all episodes.") {
+      makePageForEpisodes(allEpisodes, sectionEl);
+    } else {
+      let singleArr = [];
+      singleArr.push(allEpisodes[this.value]);
+      makePageForEpisodes(singleArr, sectionEl);
+    }
+  });
+}
+
+function makePageForEpisodes(episodeList, sectionEl) {
+  // Clear screen
+  while (sectionEl.firstChild) {
+    sectionEl.removeChild(sectionEl.firstChild);
```

Comment:
Hi Chezlin your code looks great and it seems like you have the hang of this. Just a note that if you could also do a for loop to create your elements for each episode in the makePageForEpisodes function